src/conn/conn_api.c

```diff
@@ -663,6 +663,123 @@ __wt_conn_remove_extractor(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __conn_add_shared_storage --
+ *     WT_CONNECTION->add_shared_storage method.
+ */
+static int
+__conn_add_shared_storage(
+  WT_CONNECTION *wt_conn, const char *name, WT_SHARED_STORAGE *shared_storage, const char *config)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_NAMED_SHARED_STORAGE *nshared_storage;
+    WT_SESSION_IMPL *session;
+
+    nshared_storage = NULL;
+
+    conn = (WT_CONNECTION_IMPL *)wt_conn;
+    CONNECTION_API_CALL(conn, session, add_shared_storage, config, cfg);
+    WT_UNUSED(cfg);
+
+    WT_ERR(__wt_calloc_one(session, &nshared_storage));
+    WT_ERR(__wt_strdup(session, name, &nshared_storage->name));
+    nshared_storage->shared_storage = shared_storage;
+
+    __wt_spin_lock(session, &conn->api_lock);
+    TAILQ_INSERT_TAIL(&conn->shstorageqh, nshared_storage, q);
+    nshared_storage = NULL;
+    __wt_spin_unlock(session, &conn->api_lock);
+
+err:
+    if (nshared_storage != NULL) {
+        __wt_free(session, nshared_storage->name);
+        __wt_free(session, nshared_storage);
+    }
+
+    API_END_RET_NOTFOUND_MAP(session, ret);
+}
+
+/*
+ * __shared_storage_confchk --
+ *     Check for a valid custom shared_storage.
+ */
+static int
+__shared_storage_confchk(
+  WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cname, WT_SHARED_STORAGE **shared_storagep)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_NAMED_SHARED_STORAGE *nshared_storage;
+
+    *shared_storagep = NULL;
+
+    conn = S2C(session);
+    TAILQ_FOREACH (nshared_storage, &conn->shstorageqh, q)
+        if (WT_STRING_MATCH(nshared_storage->name, cname->str, cname->len)) {
+            *shared_storagep = nshared_storage->shared_storage;
+            return (0);
+        }
+    WT_RET_MSG(session, EINVAL, "unknown shared_storage '%.*s'", (int)cname->len, cname->str);
+}
+
+/*
+ * __wt_shared_storage_config --
+ *     Given a configuration, configure the shared_storage.
+ */
+int
+__wt_shared_storage_config(
```

Comment:
I first looked at this because it seems like it collides or overlaps what the tiered storage config parsing does that I added, once it is renamed. Then I wondered who's calling it that it needs `_wt`. I cannot find any caller of this function. Is it needed? If it is needed, can we call it `wt_tiered_storage_ext_config`?

Reply:
Right - it's not needed, and the configuration check function is not currently needed.  I'm removing them to keep this clean, but a function to look up the storage source by name will be needed at some point.